great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,222 @@
+
+    __config_class__ = DomainBuilderConfig
+
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.domain_builder",
+    )
+    batch_request = fields.Dict(keys=fields.String(), required=False, allow_none=True)
+
+
+@dataclass(frozen=True)
+class ParameterBuilderConfig(DictDot):
+    name: str
+    class_name: str
+    module_name: Optional[str] = None
+    batch_request: Optional[Dict[str, Any]] = None
+
+
+class ParameterBuilderConfigSchema(NotNullSchema):
+    class Meta:
+        unknown = INCLUDE
+
+    __config_class__ = ParameterBuilderConfig
+
+    name = fields.String(required=True)
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.parameter_builder",
+    )
+    batch_request = fields.Dict(keys=fields.String(), required=False, allow_none=True)
+
+
+@dataclass(frozen=True)
+class ExpectationConfigurationBuilderConfig(DictDot):
+    expectation_type: str
+    class_name: str
+    module_name: Optional[str] = None
+    mostly: Optional[float] = None
+    meta: Optional[Dict] = None
+
+
+class ExpectationConfigurationBuilderConfigSchema(NotNullSchema):
+    class Meta:
+        unknown = INCLUDE
+
+    __config_class__ = ExpectationConfigurationBuilderConfig
+
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.expectation_configuration_builder",
+    )
+    expectation_type = fields.String(required=True)
+    mostly = fields.Float(required=False, allow_none=True)
+    meta = fields.Dict(required=False, allow_none=True)
+
+
+@dataclass(frozen=True)
+class RuleConfig(DictDot):
+    name: str
+    domain_builder: DomainBuilderConfig
+    parameter_builders: List[ParameterBuilderConfig]
+    expectation_configuration_builders: List[ExpectationConfigurationBuilderConfig]
+
+
+class RuleConfigSchema(NotNullSchema):
+    class Meta:
+        unknown = INCLUDE
+
+    __config_class__ = RuleConfig
+
+    name = fields.String(required=True)
+    domain_builder = fields.Nested(DomainBuilderConfigSchema, required=True)
+    parameter_builders = fields.List(
+        cls_or_instance=fields.Nested(ParameterBuilderConfigSchema, required=True),
+        required=True,
+    )
+    expectation_configuration_builders = fields.List(
+        cls_or_instance=fields.Nested(
+            ExpectationConfigurationBuilderConfigSchema, required=True
+        ),
+        required=True,
+    )
+
+
+@dataclass(frozen=True)
+class RuleBasedProfilerConfig(BaseYamlConfig):
+    name: str
+    config_version: float
+    rules: Dict[str, RuleConfig]
+    variables: Optional[Dict[str, Any]] = None
+    commented_map: InitVar[Optional[CommentedMap]] = None
```

Comment:
InitVar let's us pass in an arg but have it go to `__post_init__` for processing (no automatic assignment). We can use this to set a private attr as noted in L192

Reply:
It's so often that I learn something while reading your code. Thanks for this @cdkini